### PR TITLE
sinkv2(cdc): fix some races of source manager

### LINKED ISSUE #7983: Panic on cdc/processor/sourcemanager/engine/pebble.OpenPebble
### What did you do?

Run integration tests
https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_integration_test/detail/cdc_ghpr_integration_test/17672/pipeline

### What did you expect to see?

No panic.

### What did you see instead?

Panic.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x503e076]

goroutine 731 [running]:
github.com/cockroachdb/pebble.Open.func1()
	/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/open.go:112 +0x21e
panic({0x70bf360, 0xb903240})
	/usr/local/go/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/pebble/internal/manifest.(*L0Sublevels).InUseKeyRanges(0x0, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})
	/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/internal/manifest/l0_sublevels.go:887 +0x76
github.com/cockroachdb/pebble.calculateInuseKeyRanges(0xc000c7e3c0, 0x788b6e8, 0x0, 0x6, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})
	/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/compaction.go:664 +0xda
github.com/cockroachdb/pebble.(*compaction).setupInuseKeyRanges(0xc0007c82c0)
	/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/compaction.go:648 +0x1c5
github.com/cockroachdb/pebble.newFlush(0xc0007e6240, 0xc000c7e3c0, 0x3f?, {0x0, 0x0, 0x0}, 0xc0007ec010)
	/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/compaction.go:639 +0x1053
github.com/cockroachdb/pebble.(*DB).replayWAL(0xc0007ec000, 0xc0007c6600?, 0xc002ee1980, {0x93a7568, 0xc0007c60c0}, {0xc0007c4a00, 0x3f}, 0x2, 0x0)
	/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/open.go:680 +0x191d
github.com/cockroachdb/pebble.Open({0xc0007c4240, 0x34}, 0xc0007e6000)
	/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/open.go:346 +0x5199
github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.OpenPebble(0x9e911a9?, {0xc0005c4450, 0x2f}, 0x1ad5017?, 0x96ecb2cc, {0xc002ee1d20, 0x1, 0xc00166e400?})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/db.go:111 +0x634
github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory.createPebbleDBs({0xc0005c4450, 0x2f}, 0xc00045ec40, 0x4b7659666)
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory/pebble.go:60 +0x3d7
github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory.(*SortEngineFactory).Create(0xc0006b87e0, {{0xc000b20592?, 0xc0007ce280?}, {0xc000b205aa?, 0x2?}})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory/factory.go:79 +0x2b4
github.com/pingcap/tiflow/cdc/processor.(*processor).lazyInitImpl(0xc002ec4000, {0x938a2c0, 0xc0008c3d70})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:843 +0xf92
github.com/pingcap/tiflow/cdc/processor.(*processor).tick(0xc002ec4000, {0x938a2c0?, 0xc0008c3d70})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:712 +0x122
github.com/pingcap/tiflow/cdc/processor.(*processor).Tick(0xc002ec4000, {0x938a2c0, 0xc0008c3d70})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:640 +0x4ec
github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick(0xc0028a8cd0, {0x7fb640d41210?, 0xc00072a940}, {0x934caa0?, 0xc000643260})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/manager.go:142 +0x989
github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run(0xc00298c840, {0x7fb640d41210, 0xc00072a940}, 0xc00004ab10, 0x2faf080, {0x76d24d8, 0x9})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290 +0x1627
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker(0xc00143cfc0, {0x938a360?, 0xc00072a940}, {0x7fb640740008, 0xc0028a8cd0}, {0x934caa0, 0xc000643260}, 0xc0001b7c00?, {0x76d24d8, 0x9})
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:516 +0x226
github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3()
	/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:370 +0x356
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x87
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0x12f
```

And there is an error log

```
[2022/12/27 13:55:02.945 +08:00] [ERROR] [processor.go:693] ["run processor failed"] [capture=3ef2c9bc-b388-4095-a06b-44e4461bceb6] [namespace=default] [changefeed=changefeed-test1] [error="rename /tmp/tidb_cdc_test/http_api/cdc_data/tmp/sorter/0000/temporary.000003.dbtmp /tmp/tidb_cdc_test/http_api/cdc_data/tmp/sorter/0000/OPTIONS-000003: no such file or directory"] [errorVerbose="rename /tmp/tidb_cdc_test/http_api/cdc_data/tmp/sorter/0000/temporary.000003.dbtmp /tmp/tidb_cdc_test/http_api/cdc_data/tmp/sorter/0000/OPTIONS-000003: no such file or directory\n(1) attached stack trace\n  -- stack trace:\n  | github.com/cockroachdb/pebble/vfs.defaultFS.Rename\n  | \t/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/vfs/vfs.go:189\n  | github.com/cockroachdb/pebble.Open\n  | \t/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220415182917-06c9d3be25b3/open.go:433\n  | github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble.OpenPebble\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/pebble/db.go:111\n  | github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory.createPebbleDBs\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory/pebble.go:60\n  | github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory.(*SortEngineFactory).Create\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sourcemanager/engine/factory/factory.go:79\n  | github.com/pingcap/tiflow/cdc/processor.(*processor).lazyInitImpl\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:843\n  | github.com/pingcap/tiflow/cdc/processor.(*processor).tick\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:712\n  | github.com/pingcap/tiflow/cdc/processor.(*processor).Tick\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:640\n  | github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/manager.go:142\n  | github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290\n  | github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:516\n  | github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3\n  | \t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:370\n  | golang.org/x/sync/errgroup.(*Group).Go.func1\n  | \t/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75\n  | runtime.goexit\n  | \t/usr/local/go/src/runtime/asm_amd64.s:1594\nWraps: (2) rename /tmp/tidb_cdc_test/http_api/cdc_data/tmp/sorter/0000/temporary.000003.dbtmp /tmp/tidb_cdc_test/http_api/cdc_data/tmp/sorter/0000/OPTIONS-000003\nWraps: (3) no such file or directory\nError types: (1) *withstack.withStack (2) *os.LinkError (3) syscall.Errno\ngithub.com/pingcap/errors.AddStack\n\t/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/cdc/processor.(*processor).lazyInitImpl\n\t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:850\ngithub.com/pingcap/tiflow/cdc/processor.(*processor).tick\n\t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:712\ngithub.com/pingcap/tiflow/cdc/processor.(*processor).Tick\n\t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:640\ngithub.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick\n\t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/manager.go:142\ngithub.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run\n\t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n\t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:516\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3\n\t/home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:370\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1594"]
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

TiCDC version (execute `cdc version`):

```console
master 2022-12-27
```

### LINKED ISSUE #7986: DATA RACE on sinkmanager.(*redoWorker).handleTask()
### What did you do?

Run integration tests.

https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_integration_test/detail/cdc_ghpr_integration_test/17679/pipeline

### What did you expect to see?

No data race.

### What did you see instead?

```
==================
WARNING: DATA RACE
Write at 0x00c00637a000 by goroutine 1512:
  runtime.slicecopy()
      /usr/local/go/src/runtime/slice.go:307 +0x0
  github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*redoWorker).handleTask()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sinkmanager/redo_log_worker.go:276 +0x1091
  github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*redoWorker).handleTasks()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sinkmanager/redo_log_worker.go:64 +0x152
  github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*SinkManager).startWorkers.func2()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sinkmanager/manager.go:205 +0x124

Previous read at 0x00c00637a000 by goroutine 1066:
  github.com/pingcap/tiflow/cdc/redo.(*ManagerImpl).bgUpdateLog()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/redo/manager.go:716 +0x17a4
  github.com/pingcap/tiflow/cdc/redo.NewManager.func1()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/redo/manager.go:296 +0x64

Goroutine 1512 (running) created at:
  github.com/pingcap/tiflow/cdc/processor/sinkmanager.(*SinkManager).startWorkers()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sinkmanager/manager.go:203 +0x406
  github.com/pingcap/tiflow/cdc/processor/sinkmanager.New()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/sinkmanager/manager.go:159 +0xa4b
  github.com/pingcap/tiflow/cdc/processor.(*processor).lazyInitImpl()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:853 +0x1a44
  github.com/pingcap/tiflow/cdc/processor.(*processor).lazyInitImpl-fm()
      <autogenerated>:1 +0x4d
  github.com/pingcap/tiflow/cdc/processor.(*processor).tick()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:712 +0x121
  github.com/pingcap/tiflow/cdc/processor.(*processor).Tick()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:640 +0x4eb
  github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/manager.go:142 +0x988
  github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290 +0x1626
  github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:516 +0x225
  github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:370 +0x355
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x86

Goroutine 1066 (running) created at:
  github.com/pingcap/tiflow/cdc/redo.NewManager()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/redo/manager.go:296 +0x16c9
  github.com/pingcap/tiflow/cdc/processor.(*processor).lazyInitImpl()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:833 +0xc2d
  github.com/pingcap/tiflow/cdc/processor.(*processor).lazyInitImpl-fm()
      <autogenerated>:1 +0x4d
  github.com/pingcap/tiflow/cdc/processor.(*processor).tick()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:712 +0x121
  github.com/pingcap/tiflow/cdc/processor.(*processor).Tick()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/processor.go:640 +0x4eb
  github.com/pingcap/tiflow/cdc/processor.(*managerImpl).Tick()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/processor/manager.go:142 +0x988
  github.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:290 +0x1626
  github.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:516 +0x225
  github.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3()
      /home/jenkins/agent/workspace/cdc_ghpr_integration_test/go/src/github.com/pingcap/tiflow/cdc/capture/capture.go:370 +0x355
  golang.org/x/sync/errgroup.(*Group).Go.func1()
      /go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x86
==================
```

### Versions of the cluster

TiCDC version (execute `cdc version`):

```console
master 2022-12-28
```

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: close #7986
Issue Number: close #7983 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sinkmanager/redo_log_worker.go [hunk 1]
```diff
@@ -116,8 +116,9 @@ func (w *redoWorker) handleTask(ctx context.Context, task *redoTask) (finalErr e
 						zap.Uint64("memory", refundMem))
 				}
 			}
+			sendRows := rows[0:]
```
Comment (overvenus) [inline cdc/processor/sinkmanager/redo_log_worker.go:119]: They still shares the same underlaying memory, I am not sure if it fixes the race.

**Review by amyangfei — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**